File: src/main/java/me/anviks/_7_kyu/list_filtering/ExampleTests.java
package me.anviks._7_kyu.list_filtering;

/*
 * https://www.codewars.com/kata/53dbd5315a3c69eed20002dd
 */

import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class ExampleTests {

    @Test
    public void examples() {
        assertEquals(List.of(1, 2), Kata.filterList(List.of(1, 2, "a", "b")), "For input: [1, 2, \"a\", \"b\"]");
        assertEquals(List.of(1, 0, 15), Kata.filterList(List.of(1, "a", "b", 0, 15)), "For input: [1, \"a\", \"b\", 0, 15]");
        assertEquals(List.of(1, 2, 123), Kata.filterList(List.of(1, 2, "aasf", "1", "123", 123)), "For input: [1, 2, \"aasf\", \"1\", \"123\", 123]");
    }
}

File: src/main/kotlin/me/anviks/codewars/translations/ExampleTests.kt
//package me.anviks.codewars.translations
//
//import org.junit.jupiter.api.Test
//import org.junit.jupiter.api.DisplayName
//import org.junit.jupiter.api.Assertions.assertEquals
//
//class SampleTests {
//    @Test
//    @DisplayName("Should return true for valid parentheses")
//    fun validCases() {
//        runTest(true, "()")
//        runTest(true, "((()))")
//        runTest(true, "()()()")
//        runTest(true, "(()())()")
//        runTest(true, "()(())((()))(())()")
//    }
//
//    @Test
//    @DisplayName("Should return false for invalid parentheses")
//    fun invalidCases() {
//        runTest(false, ")(")
//        runTest(false, "()()(")
//        runTest(false, "((())")
//        runTest(false, "())(()")
//        runTest(false, ")()")
//        runTest(false, ")")
//    }
//
//    @Test
//    @DisplayName("Should return true for empty strings")
//    fun emptyString() {
//        runTest(true, "")
//    }
//
//    private fun runTest(expected: Boolean, input: String) {
//        assertEquals(expected, validParentheses(input), "Test failed for input \"$input\"")
//    }
//}
//
